pandas/indexes/numeric.py

```diff
@@ -90,29 +116,7 @@ class Int64Index(NumericIndex):
 
     _engine_type = _index.Int64Engine
 
-    def __new__(cls, data=None, dtype=None, copy=False, name=None,
-                fastpath=False, **kwargs):
-
-        if fastpath:
-            return cls._simple_new(data, name=name)
-
-        # isscalar, generators handled in coerce_to_ndarray
-        data = cls._coerce_to_ndarray(data)
-
-        if issubclass(data.dtype.type, compat.string_types):
-            cls._string_data_error(data)
-
-        elif issubclass(data.dtype.type, np.integer):
-            dtype = np.int64
-            subarr = np.array(data, dtype=dtype, copy=copy)
-        else:
-            subarr = np.array(data, dtype=np.int64, copy=copy)
-            if len(data) > 0:
-                if (subarr != data).any():
-                    raise TypeError('Unsafe NumPy casting to integer, you must'
-                                    ' explicitly cast')
```

Comment:
oh instead of ``TypeError``. Hmm, we might be a bit sloppy on this. But we DO raise ``TypeError`` (e.g. in case of Int64Index -> DatetimeIndex), so this is prob right (TypeError). 

Idea is that is a bit more 'specific' in that its a type related operation. This is a bit subjective, but let's leave this alone.